MarkBind/markbind
# Fix hasPageNav logic to only apply data-bs attributes if there are navigable headings

**What is the purpose of this pull request?**




**Overview of changes:**
Fixes #1954

**Anything you'd like to highlight/discuss:**
nil

**Testing instructions:**
- check functional test written
- check reproduction steps in the original issue no longer generates an error when running `markbind serve -d`



**Proposed commit message: (wrap lines at 72 characters)**
Fix hasPageNav logic

Bootstrap data-bs attributes are applied to the body element when
there is pageNav specified in the frontmatter and the layout.

Let's fix the hasPageNav logic to account for the fact that if there 
are no navigable headings, the Bootstrap attributes are not needed.

This removes the read null error in the console.



---

**Checklist:** :ballot_box_with_check:
